tests/actions/ReportTest.js

```diff
@@ -0,0 +1,102 @@
+import moment from 'moment';
+import Onyx from 'react-native-onyx';
+import ONYXKEYS from '../../src/ONYXKEYS';
+import * as Pusher from '../../src/libs/Pusher/pusher';
+import PusherConnectionManager from '../../src/libs/PusherConnectionManager';
+import CONFIG from '../../src/CONFIG';
+import {addAction, subscribeToReportCommentEvents} from '../../src/libs/actions/Report';
+import waitForPromisesToResolve from '../utils/waitForPromisesToResolve';
+import PushNotification from '../../src/libs/Notification/PushNotification';
+import {signInWithTestUser, fetchPersonalDetailsForTestUser} from '../utils/TestHelper';
+
+PushNotification.register = () => {};
+PushNotification.deregister = () => {};
+
+describe('actions/Report', () => {
+    it('should store a new report action in Onyx when one is handled via Pusher', () => {
+        const TEST_USER_ACCOUNT_ID = 1;
+        const TEST_USER_LOGIN = 'test@test.com';
+        const REPORT_ID = 1;
+        const ACTION_ID = 1;
+        const REPORT_ACTION = {
+            actionName: 'ADDCOMMENT',
+            actorAccountID: TEST_USER_ACCOUNT_ID,
+            actorEmail: TEST_USER_LOGIN,
+            automatic: false,
+            avatar: 'https://d2k5nsl2zxldvw.cloudfront.net/images/avatars/avatar_3.png',
+            isAttachment: false,
+            isFirstItem: false,
+            loading: true,
+            message: [{type: 'COMMENT', html: 'Testing a comment', text: 'Testing a comment'}],
+            person: [{type: 'TEXT', style: 'strong', text: 'Test User'}],
+            sequenceNumber: ACTION_ID,
+            shouldShow: true,
+            timestamp: moment().unix(),
+        };
+
+        // When using the Pusher mock the act of calling Pusher.isSubscribed will create a
+        // channel already in a subscribed state. These methods are normally used to prevent
+        // duplicated subscriptions, but we don't need them for this test.
```

Comment:
```suggestion
        // duplicated subscriptions, but we don't need them for this test so forcing them to return false will make the testing less complex.
```
This adds a little more clarity.